File: Java/DecoratorPattern/src/addons/Milk.java
package addons;

import coffee.Coffee;

/**
 * Author: Utkarsh Dayal @ descifrador17
 * Date: 27-05-2021
 * Time: 04:26 AM
 */
public class Milk extends Addons{

    Coffee coffee;

    double milkCost = 5.0;

    public Milk(Coffee coffee){
        this.coffee = coffee;
        getDescription();
    }

    @Override
    public double cost() {
        return this.coffee.cost()+milkCost;
    }

    @Override
    public String getDescription() {
        return coffee.getDescription()+" + Milk : Rs "+milkCost;
    }
}

File: Kotlin/DecoratorPattern/src/addons/Milk.kt
package addons

import coffee.Coffee

/**
 *
 *   Author: Utkarsh Dayal @ descifrador17
 *   Date: 27-05-2021
 *   Time: 04:57 AM
 *
 */
class Milk(coffee: Coffee) : Addons(){

    private val c = coffee

    var milkCost = 5.0

    override fun getCoffeeDescription(): String {
        return "${c.getCoffeeDescription()} + Milk : Rs $milkCost"
    }

    override fun cost(): Double {
        return c.cost()+milkCost
    }


}